### PR TITLE
 bind enforcement to validate phase

### PR DESCRIPTION
**Pull request by elharo**
Fails locally:

[ERROR] Failed to execute goal on project google-cloud-core-grpc: Could not resolve dependencies for project com.google.cloud:google-cloud-core-grpc:jar:1.27.1-SNAPSHOT: Could not find artifact com.google.cloud:google-cloud-core:jar:1.27.1-SNAPSHOT -> [Help 1]
[ERROR]

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -502,7 +503,8 @@
                 <requireJavaVersion>
                   <version>[1.7,)</version>
                 </requireJavaVersion>
-                <dependencyConvergence/>
+                <dependencyConvergence>
+                </dependencyConvergence>
```
Comment (garrettjonesgoogle) [inline pom.xml]: Why are you breaking this apart into start/end tags?

Comment (elharo) [reply pom.xml]: No particular reason. It's a random change that happened as I was experimenting with different options to try to make this work.

**Comment by sduskis** [comment]
@elharo, do you mind resolving the conflicts?

**Comment by elharo** [comment]
Tried to merge, but please check carefully. A lot more came in than I expected.

**Comment by sduskis** [comment]
@garrettjonesgoogle and @chingor13, do either of you have any objections to this PR?

File: pom.xml [hunk 2]
```diff
@@ -104,6 +104,106 @@
...
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.19.1</version>
+        <configuration>
+          <!-- Excludes integration tests and smoke tests when unit tests are run -->
+          <excludes>
+            <exclude>**/*SmokeTest.java</exclude>
+            <exclude>**/IT*.java</exclude>
+            <exclude>**/*SmokeTest.java</exclude>
```
Comment (garrettjonesgoogle) [inline pom.xml]: This exclusion appears twice

File: pom.xml [hunk 3]
```diff
@@ -104,6 +104,106 @@
...
+            <configuration>
+              <rules>
+                <requireMavenVersion>
+                  <version>[3.0,)</version>
+                </requireMavenVersion>
+                <requireJavaVersion>
+                  <version>[1.7,)</version>
+                </requireJavaVersion>
+                <dependencyConvergence />
+                <requireUpperBoundDeps/>
```
Comment (garrettjonesgoogle) [inline pom.xml]: I thought we already had this enforcement on somewhere

File: pom.xml [hunk 4]
```diff
@@ -104,6 +104,106 @@
...
+            </goals>
+            <configuration>
+              <rules>
+                <requireMavenVersion>
+                  <version>[3.0,)</version>
+                </requireMavenVersion>
+                <requireJavaVersion>
+                  <version>[1.7,)</version>
+                </requireJavaVersion>
+                <dependencyConvergence />
```
Comment (garrettjonesgoogle) [inline pom.xml]: remove the extra space inside the tag

File: pom.xml [hunk 5]
```diff
@@ -104,6 +104,106 @@
...
+                <requireUpperBoundDeps/>
+              </rules>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
```
Comment (garrettjonesgoogle) [inline pom.xml]: What do we need this plugin for?

File: pom.xml [hunk 6]
```diff
@@ -104,6 +104,106 @@
...
+        <executions>
+          <execution>
+            <goals>
+              <goal>test-jar</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
```
Comment (garrettjonesgoogle) [inline pom.xml]: Why do we need to add this?

File: pom.xml [hunk 7]
```diff
@@ -104,6 +104,106 @@
...
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>2.6</version>
+        <configuration>
+          <archive>
+            <addMavenDescriptor>true</addMavenDescriptor>
```
Comment (garrettjonesgoogle) [inline pom.xml]: What is a maven descriptor?

**Comment by sduskis** [comment]
@elharo, it looks like @garrettjonesgoogle has some questions about what the goal is.  Can you please elaborate?

**Comment by chingor13** [comment]
I'm going to close this out as it's unclear what this is trying to fix. Feel free to reopen with more information on the goal.

**Pull request closed without merging**